src/item_location.cpp

```diff
@@ -31,7 +33,12 @@ class item_location::impl
         virtual void remove_item() = 0;
 
     protected:
-        item *what = nullptr;
+        impl( item_uid &&id ) : id( std::move( id ) ) {}
```

Comment:
This was discussed similarly at the last code review. I'd prefer the class to worry about implementation details than the consumer. I can add a code comment though.